File: jaeseuk/pypython/Programmers/level2/H-Index.py
def solution(citations):
    citations.sort()
    answer = citations[-1]

    while answer >= 0:
        up = 0
        down = 0
        for citation in citations:
            if citation >= answer:
                up += 1
            elif citation <= answer:
                down += 1

        if up >= answer >= down:
            break

        answer -= 1

    return answer

File: yoongyeong/by_kotlin/src/main/kotlin/programmers/level2/HIndex.kt
package programmers.level2

class HIndex {
    fun solution(citations: IntArray): Int {
        val sortCitations = citations.sortedDescending()
        for ((i, citation) in sortCitations.withIndex()) {
            if (citation <= i) {
                return i
            }
        }
        return citations.size
    }
}